### ISSUE TITLE
Boost dependency

### ISSUE THREAD

**Issue by vchernetsky** [socketio/socket.io-client-cpp#124]
Hello socket.io developers, this is just a general wondering. Are you guys planning in some near future to remove dependencies on Boost? The problem for me, and I'm sure for other people, is that Boost dependency makes this wonderful library not portable. I believe with latest advances in C++-11 there is no real reason to depend on Boost. Thank you.

**Comment by akontsevich** [comment]
Agree, with boost app deployment becomes much more complicated, especially to mobile or embedded devices. However is it built on ASIO then boost is critical here. Is it possible to remove this dependency?
